homeassistant/components/velbus/config_flow.py

```diff
@@ -46,25 +48,74 @@ async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Step when user initializes a integration."""
-        self._errors = {}
+        return self.async_show_menu(
+            step_id="user", menu_options=["network", "usbselect"]
+        )
+
+    async def async_step_network(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle network step."""
+        if user_input is not None:
+            self._title = "Velbus Network"
+            if user_input[CONF_TLS] == "yes":
+                self._device = "tls://"
+            else:
+                self._device = ""
+            if user_input[CONF_PASSWORD] != "":
+                self._device += f"{user_input[CONF_PASSWORD]}@"
+            self._device += f"{user_input[CONF_HOST]}:{user_input[CONF_PORT]}"
+            self._async_abort_entries_match({CONF_PORT: self._device})
```

Comment:
considered using a library like YARL to piece the puzzle of the URL back together?

Reply:
yarl can't build a string like we need it, yarl always adds the // at the beginning, even if we do not set a scheme.
so it will always start with //
so we have to strip it if there is no scheme provided, not sure if this makes the code more readable

Reply:
yarl will always add // even if no scheme is passed, so i can't make it work

Reply:
But what is the scheme if tls is not used?

Reply:
velbus-aio takes the following for network

tls://user@host:port or host:port

yarl will build the first one, but the second one yarl will always add the // in front

Reply:
Yes, but I mean, what does velbus-aio use internally if we don't provide `tls://`?

Reply:
What does it use to connect? Wouldn't it make sense to just throw all the parameters to the library and have that figure it out?

Reply:
in the library we also expect this string, so this requires a library change.
Plus in the lib i will need to support both, as there is another project also using this lib that has nothing todo with home-assistant.